### PR TITLE
:sparkles: reset a given path

### LINKED ISSUE #72: Add a reset store function
**In different use cases, sometimes, I need to reset the store or a part of the store.**

For example, when I switch from a connected user to another I need to reset a huge part of the store and its quite exhaustive to do. Because I have to call each `reset()` methods of my reducers, or create `RESET` actions to my custom reducers. And, in the future, if I have to add a new reducer, I dont must forget to reset it when needed.

**I was thinking about a method :** 
- `store.reset()` : apply the initial state to the entire store
- `store.reset('path')` : apply the initial state from the given path

### PR DESCRIPTION
**Pull request by fabienjuif**
fixes #72 

### TODO
 - [x] Tests
   * krf types
   * custom reducer
 - [x] Documentation
 - [x] Add type constant: `RESET` (like in krf)

### CODE REVIEW

**Comment by fabienjuif** [comment]
I added the type to reset.TYPE, but I should add it to `RESET` if we want this issue to work: https://github.com/alakarteio/k-ramel/issues/297

File: packages/k-ramel/src/createStore/addReset.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  const addReset = addResetFactory(options, store.dispatch)
+  const subcontext = (name, path) => {
+    // first run
+    if (name === undefined) {
+      return Object
+        .keys(root)
+        .map(key => ({ [key]: subcontext(key, '') }))
+        .reduce(
+          (acc, next) => ({ ...acc, ...next }),
+          {
```
Comment (guillaumecrespel) [inline packages/k-ramel/src/createStore/addReset.js]:
```suggestion
          {},
```

File: packages/k-ramel/src/createStore/addReset.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          },
```
Comment (guillaumecrespel) [inline packages/k-ramel/src/createStore/addReset.js]:
```suggestion
```

**Comment by fabienjuif** [comment]
merged via #310

**Pull request closed without merging**